<launch>
    <!-- Read robot config -->
    <rosparam file="$(find breach)/config/robot_config.yaml" command="load"/>

    <!-- Read robot model -->
    <arg name="model" default="$(find breach)/urdf/breach.urdf" />
    <arg name="gui" default="False" />
    <param name="robot_description" textfile="$(arg model)" />
    <param name="use_gui" value="$(arg gui)"/>

    <!-- State publisher, tf -->
    <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
        <param name="rate" value="10"/>
    </node>
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher">
        <param name="publish_frequency" value="10"/>
    </node>

    <!-- Serial and odometry -->
    <node name="control" pkg="breach" type="serialReceive" output="screen" />
    <node name="odom" pkg="breach" type="odometry_publisher" />

    <!-- RP Lidar -->
    <node name="urg_node" pkg="urg_node" type="urg_node" output="screen">
        <param name="ip_address" value=""/>
        <param name="serial_port" value="/dev/ttyACM0"/>
        <param name="serial_baud" value="115200"/>
        <param name="frame_id" value="laser"/>
        <param name="calibrate_time" value="true"/>
        <param name="publish_intensity" value="false"/>
        <param name="publish_multiecho" value="false"/>
        <param name="angle_min" value="-2.094395102"/>
        <param name="angle_max" value="2.094395102"/>
    </node>

    <!-- ROSBAG -->
    <node pkg="rosbag" type="record" name="rosbag_record"
       args="-o bag/breach/core
            /breach/voltage /breach/current
            /breach/error/estop /breach/error/bumper_front
                /breach/error/bumper_rear /breach/error/battery
            /move_base/global_costmap/footprint /move_base/goal
                /move_base/local_costmap/costmap
            /move_base/local_costmap/footprint /move_base/result /move_base/status
            /amcl_pose
            /scan
            /srf01 /srf02 /srf03 /srf04 /srf05 /srf06 /srf07 /srf08 /srf09 /srf10
            /tf /tf_static /joint_states /odom
            /map_metadata /map
            /error /diagnostics /rosout
            "/>
</launch>
